core/peer/config.go

```diff
@@ -274,7 +274,12 @@ func (c *Config) load() error {
 	}
 
 	c.GatewayOptions = gateway.DefaultOptions()
-	c.GatewayOptions.Enabled = viper.GetBool("peer.gateway.enabled")
+	if viper.IsSet("peer.gateway.enabled") {
```

Comment:
Since this is new code, the logic to create the gateway configuration should live in the gateway code. If possible (and I think we're finally at a level of the code that supports it), the viper instance should be passed to the constructor to avoid dealing with global state.